Web/Edubase.Services.TexunaUnitTests/Establishments/EstablishmentReadApiServiceTests.cs

```diff
@@ -0,0 +1,714 @@
+using System;
+using System.Collections.Generic;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Edubase.Services.Domain;
+using Edubase.Services.Establishments.DisplayPolicies;
+using Edubase.Services.Establishments.EditPolicies;
+using Edubase.Services.Establishments.Models;
+using Edubase.Services.Establishments.Search;
+using Edubase.Services.Lookup;
+using Edubase.Services.Texuna.Establishments;
+using Edubase.Services.Texuna.Models;
+using Moq;
+using Xunit;
+
+namespace Edubase.Services.TexunaUnitTests.Establishments
+{
+    public class EstablishmentReadApiServiceTests
+    {
+        private readonly Mock<IHttpClientWrapper> _mockHttpClientWrapper = new Mock<IHttpClientWrapper>();
+        private readonly Mock<ICachedLookupService> _mockCachedLookupService = new Mock<ICachedLookupService>();
+
+        public EstablishmentReadApiServiceTests()
+        {
+        }
+
+        [Fact]
+        public async Task CanAccess_Returns_True_For_Successful_Response_From_Api()
+        {
+            //Arrange
+            var apiResponse = new ApiResponse<BoolResult>(true) { Response = new BoolResult() { Value = true } };
+
+            _mockHttpClientWrapper.Setup(x => x.GetAsync<BoolResult>(It.IsAny<string>(), It.IsAny<IPrincipal>()))
+                .ReturnsAsync(() => apiResponse);
+
+            //Act
+            var service = GetEstablishmentReadApiService();
+            var result = await service.CanAccess(It.IsAny<int>(), null);
```

Comment:
Don't use the It.IsAny<int> here - this is just for Mock Objects

Reply:
Also lines 56, 91

Reply:
@ScottDawsonDfE ok, updated